src/main/java/no/sikt/nva/scrapers/DublinCoreOnlineValidator.java

```diff
@@ -0,0 +1,73 @@
+package no.sikt.nva.scrapers;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import no.sikt.nva.model.ErrorDetails;
+import no.sikt.nva.model.ErrorDetails.Error;
+import no.sikt.nva.model.dublincore.DcValue;
+import no.sikt.nva.model.dublincore.DublinCore;
+import nva.commons.doi.DoiValidator;
+import nva.commons.doi.UnitHttpClient;
+import org.jetbrains.annotations.NotNull;
+
+public class DublinCoreOnlineValidator {
+
+    public static final String HTTP_STRING = "http";
+    public static final String HTTPS_STRING_FOR_DOI = "https://";
+
+    public static List<ErrorDetails> getDublinCoreErrors(DublinCore dublinCore) {
+
+        var errors = new ArrayList<ErrorDetails>();
+        getDoiErrorDetails(dublinCore).ifPresent(errors::add);
+        return errors;
+    }
+
+    private static Optional<ErrorDetails> getDoiErrorDetails(DublinCore dublinCore) {
+        var doiList =
+            dublinCore.getDcValues()
+                .stream()
+                .filter(DcValue::isDoi)
+                .map(DcValue::getValue)
+                .map(DublinCoreOnlineValidator::addHttpStringIfNotPresent)
+                .collect(Collectors.toList());
+
+        return validateDoiList(doiList);
+    }
+
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    @NotNull
+    private static Optional<ErrorDetails> validateDoiList(List<String> doiList) {
+        if (doiList.isEmpty()) {
+            return Optional.empty();
+        } else {
+            for (String doi : doiList) {
+                if (!isValidDoiOnline(doi)) {
+                    return Optional.of(new ErrorDetails(Error.INVALID_DOI_ONLINE_CHECK, doiList));
```

Comment:
Denne metoden er merkelig. Fra koden er det veldig utydelig om en ressurs kan ha én eller flere DOI'er. For min del ser det ut som den returnerer Optional.empty hvis den ikke finner noen invalide DOI'er, men så er den merkelig fordi hvis det er to DOI'er som er invalide så vil den kun returnere den første den støter på som invalid hvis jeg forstår denne koden riktig.

Reply:
endret denne